packages/components/src/templates/next/components/complex/DynamicDataBanner/DynamicDataBannerClient.tsx

```diff
@@ -69,32 +65,37 @@ export const DynamicDataBannerUI = ({
   )
 }
 
-type DynamicDataBannerClientProps = Omit<
-  DynamicDataBannerProps,
-  "type" | "site"
->
-
-const DynamicDataBannerContent = ({
+export const DynamicDataBannerClient = ({
   apiEndpoint,
   title,
   data,
   url,
   label,
   LinkComponent,
-}: DynamicDataBannerClientProps) => {
-  const {
-    isPending,
-    error,
-    data: apiData,
-  } = useQuery({
-    queryKey: [getSingaporeDateYYYYMMDD()],
-    queryFn: async () => {
-      const response = await fetch(apiEndpoint)
-      return (await response.json()) as Record<string, object | string>
-    },
-  })
+}: Omit<DynamicDataBannerProps, "type" | "site">) => {
+  const [isLoading, setLoading] = useState(true)
+  const [apiData, setApiData] = useState<Record<string, object>>({})
+
+  // This is to ensure that the component is mounted before the query is executed
```

Comment:
did we manage to test the build?

Reply:
yup! https://test-isomer-next-staging.isomer.gov.sg/

<img width="1526" alt="image" src="https://github.com/user-attachments/assets/ec0c92a1-2cfd-4e79-8927-e09ac2e38185">

Reply:
Curious, what is the fallback state if the API returns with an error? 

Can we choose to not render this component in this case? Is that the current behaviour

Reply:
fallback state is unchanged, which means it will be the same as loading - banner is still showing but without the values) - "view all dates" button will still be available

we can don't render this whole banner if that's the case, but wondering if that's better since user unable to click "view all dates"?

Reply:
I think the current behaviour is fine for now
Cc @jandonyada

Reply:
If there's an error, can we show some text with a link to a formsg or something? Like "Not seeing the prayer times? [Report an issue](www.form.gov.sg/some-link)."

Reply:
> If there's an error, can we show some text with a link to a formsg or something? Like "Not seeing the prayer times? [Report an issue](www.form.gov.sg/some-link)."

Do we already have an existing link I can use?

Reply:
@jandonyada @harishv7 done!

https://www.chromatic.com/build?appId=665439911775ce21bf8b8fb4&number=1365